kore/src/Kore/Attribute/Concat.hs

```diff
@@ -0,0 +1,94 @@
+{-|
+Module      : Kore.Attribute.Concat
+Description : Concat attribute
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : thomas.tuegel@runtimeverification.com
+
+-}
+
+{-# LANGUAGE Strict #-}
+
+module Kore.Attribute.Concat
+    ( Concat (..)
+    , mergeConcat, toConcat, fromConcat
+    , concatId, concatSymbol, concatAttribute
+    ) where
+
+import Prelude.Kore
+
+import Data.Default
+import qualified Generics.SOP as SOP
+
+import Kore.Attribute.Parser
+import Kore.Debug
+import Kore.Unparser
+
+-- | @Concat@ represents the @concat@ attribute.
+newtype Concat symbol = Concat { getConcat :: Maybe symbol }
+    deriving (Generic, Eq, Ord, Show)
+    deriving (Foldable, Functor, Traversable)
+    deriving anyclass (Hashable)
+    deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
+    deriving anyclass (Debug, Diff)
+
+mergeConcat :: (Eq symbol , Unparse symbol)
+    => Concat symbol -> Concat symbol -> Concat symbol
+mergeConcat (Concat Nothing) b = b
+mergeConcat a (Concat Nothing) = a
+mergeConcat a@(Concat (Just aSymbol)) (Concat (Just bSymbol))
+      | aSymbol == bSymbol = a
+      | otherwise = error $
+        "Concat symbol mismatch error! Found both "
+        ++ unparseToString aSymbol
+        ++ " and "
+        ++ unparseToString bSymbol
+        ++ " inside term. This is a bug in the Haskell backend."
```

Comment:
Would it be better to add an error log entry type for this? Also, we should probably format the message with `Pretty`.

I have an additional question. If this throws an error, are we 100% certain that this is a bug in the backend or is it possible that the frontend generated the attributes wrong?

Reply:
To quickly answer your second question, it could be indeed that the frontend generated some bad attributes. I should probably change that message to just say "This is a bug." as it could be in either the frontend or backend.